lib/cocoapods/sources_manager.rb

```diff
@@ -114,6 +114,7 @@ def update_git_repo(show_output = false)
         git!(args)
         current_branch = git!(%W(-C #{repo} rev-parse --abbrev-ref HEAD)).strip
         git!(%W(-C #{repo} reset --hard origin/#{current_branch}))
+        system 'git gc --auto'
```

Comment:
`git gc --auto` only garbage collects if git [determines it is required](https://git-scm.com/docs/git-gc) - I tried running this before running `git gc --prune` and it exited without taking any action. Will this actually have any effect?